src/utils/utils.ts

```diff
@@ -17,3 +17,30 @@ export function composedPath(el) {
 
   return path;
 }
+
+// Turn location URL into breadcrumbs
+export function getBreadcrumbs() {
+  let path = "/";
+  let result = [];
+  for (let encoded_step of window.location.pathname.split('/')) {
+    let step = decodeURI(encoded_step)
+    if (result.length == 0) {
+      if (step == "search") {
+        result.push({ name: "Search", url: "/search" });
+      }
+    } else if (result.length == 1) {
+      path = "/search/search-results";
+      result.push({ name: "Search Results", url: path });
+    } else {
+      path += "/" + step;
+      result.push({ name: decodeURI(step).replace('-', ' '), url: path });
+    }
+    if (result.length == 6) {
+      result[5].data = history.state.data;
+    }
+  }
+  if (window.innerWidth <= 991 && 1 < result.length) {
+    result.shift();
+  }
+  return result;
+}
```

Comment:
I suspect there might be a slightly cleaner implementation possible here. Angular has a `router` library where it should be possible to extract the pathname and perhaps offer some out-of-the-box functionality.